pkg/metrics/metrics.go

```diff
@@ -557,72 +365,594 @@ var (
+			}, []string{"host", LabelMethod, LabelAPIReturnCode})
+
+			collectors = append(collectors, KubernetesAPICalls)
+			c.KubernetesAPICallsEnabled = true
+
+		case Namespace + "_" + SubsystemK8s + "_cnp_status_completion_seconds":
+			KubernetesCNPStatusCompletion = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+				Namespace: Namespace,
+				Subsystem: SubsystemK8s,
+				Name:      "cnp_status_completion_seconds",
+				Help:      "Duration in seconds in how long it took to complete a CNP status update",
+			}, []string{LabelAttempts, LabelOutcome})
+
+			collectors = append(collectors, KubernetesCNPStatusCompletion)
+			c.KubernetesCNPStatusCompletionEnabled = true
+
+		case Namespace + "_ipam_events_total":
+			IpamEvent = prometheus.NewCounterVec(prometheus.CounterOpts{
+				Namespace: Namespace,
+				Name:      "ipam_events_total",
+				Help:      "Number of IPAM events received labeled by action and datapath family type",
+			}, []string{LabelAction, LabelDatapathFamily})
+
+			collectors = append(collectors, IpamEvent)
+			c.IpamEventEnabled = true
+
+		case Namespace + "_" + SubsystemKVStore + "_operations_duration_seconds":
+			KVStoreOperationsDuration = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+				Namespace: Namespace,
+				Subsystem: SubsystemKVStore,
+				Name:      "operations_duration_seconds",
+				Help:      "Duration in seconds of kvstore operations",
+			}, []string{LabelScope, LabelKind, LabelAction, LabelOutcome})
+
+			collectors = append(collectors, KVStoreOperationsDuration)
+			c.KVStoreOperationsDurationEnabled = true
+
+		case Namespace + "_" + SubsystemKVStore + "_events_queue_seconds":
+			KVStoreEventsQueueDuration = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+				Namespace: Namespace,
+				Subsystem: SubsystemKVStore,
+				Name:      "events_queue_seconds",
+				Help:      "Duration in seconds of time received event was blocked before it could be queued",
+				Buckets:   []float64{.002, .005, .01, .015, .025, .05, .1, .25, .5, .75, 1},
+			}, []string{LabelScope, LabelAction})
+
+			collectors = append(collectors, KVStoreEventsQueueDuration)
+			c.KVStoreEventsQueueDurationEnabled = true
+
+		case Namespace + "_fqdn_gc_deletions_total":
+			FQDNGarbageCollectorCleanedTotal = prometheus.NewCounter(prometheus.CounterOpts{
+				Namespace: Namespace,
+				Name:      "fqdn_gc_deletions_total",
+				Help:      "Number of FQDNs that have been cleaned on FQDN Garbage collector job",
+			})
+
+			collectors = append(collectors, FQDNGarbageCollectorCleanedTotal)
+			c.FQDNGarbageCollectorCleanedTotalEnabled = true
+		}
+	}
 
-	MustRegister(KVStoreOperationsDuration)
-	MustRegister(KVStoreEventsQueueDuration)
+	return c, collectors
+}
 
-	MustRegister(FQDNGarbageCollectorCleanedTotal)
+func init() {
+	MustRegister(prometheus.NewProcessCollector(prometheus.ProcessCollectorOpts{Namespace: Namespace}))
+	// TODO: Figure out how to put this into a Namespace
```

Comment:
What's the story behind this?

Reply:
It was already there. git blames @raybejjani on this one